### PR TITLE
Postcode search

### PR DESCRIPTION
**Pull request by camillebaldock**

### CODE REVIEW

File: service/server.py [hunk 1]
```diff
@@ -175,24 +178,42 @@ def find_titles():
...
-        return render_template(
-            'search.html',
-            asset_path='../static/',
-            google_api_key=GOOGLE_ANALYTICS_API_KEY,
-            form=TitleSearchForm()
-        )
+                return redirect(url_for('display_title', title_ref=search_term.upper()))
+            else:
+                # If title not found display 'no title found' screen
+                return render_template('search_results.html',
```
Comment (lgonczar) [inline service/server.py]: Now that PEP8 adjustments have made this method very long, we could think about extracting the `render_template` calls into separate functions, such as `render_search_results`.

File: service/templates/search_results.html [hunk 2]
```diff
@@ -33,8 +33,26 @@ <h1 class="heading-medium">
                               <button type="submit" class="button">Search</button>
                           </div>
                       </form>
-
+                      {% if results %}
+                        <div class="column-two-thirds">
+                          <h2 class="font-xsmall expand-bottom-half">{{ results['titles']|length }} results found</h2>
```
Comment (lgonczar) [inline service/templates/search_results.html]: How about `result(s) found` ?

File: service/server.py (hunk 1 above)
Comment (lgonczar) [inline service/server.py]: Actually, any kind of decomposition would be fine.

File: service/server.py [hunk 3]
```diff
@@ -161,13 +161,16 @@ def display_title(title_ref):
...
     if request.method == "POST":
-        search_term = request.form['search_term']
+        search_term = request.form['search_term'].strip()
         LOGGER.info(
             "SEARCH REGISTER: {0} was searched by {1}".format(
                 search_term,
                 current_user.get_id()))
         # Determine search term type and preform search
         title_number_regex = re.compile(TITLE_NUMBER_REGEX)
+        postcode_regex = re.compile(
```
Comment (lgonczar) [inline service/server.py]: The regex could be a global variable. As Python is probably going to compile it once anyway, not really important.

**Comment by lgonczar** [comment]
Reviewed. Looks good. Left some comments, just to consider. Good to be merged.

**Comment by camillebaldock** [comment]
@lgonczar's comments addressed, thanks for the quick feedback :+1:

**Pull request merged**